src/arch/x86/kernel/boot_sys.c

```diff
@@ -63,23 +63,25 @@ BOOT_CODE static paddr_t find_load_paddr(paddr_t min_paddr, word_t image_size)
     int i;
 
     for (i = 0; i < boot_state.mem_p_regs.count; i++) {
-        paddr_t start = MAX(min_paddr, boot_state.mem_p_regs.list[i].start);
+        paddr_t start = MAX(boot_state.ki_p_reg.end, MAX(min_paddr, boot_state.mem_p_regs.list[i].start));
```

Comment:
why is this change needed and correct?

Reply:
Probably a bugfix, but it should have much more explanation. It should be well defined what the end of the boot state is, now two addresses are checked, which is weird.